libs/common/src/error.rs

```diff
@@ -0,0 +1,95 @@
+//! Error module.
+use base64::{DecodeError, DecodeSliceError};
+use boringtun::noise::errors::WireGuardError;
+use macros::SwiftEnum;
+use thiserror::Error;
+
+/// Unified Result type to use across connlib.
+pub type Result<T> = std::result::Result<T, ConnlibError>;
+
+/// Unified error type to use across connlib.
+#[derive(Error, Debug, SwiftEnum)]
+pub enum ConnlibError {
+    /// Standard IO error.
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    /// Error while decoding a base64 value.
+    #[error("There was an error while decoding a base64 value: `{0}`")]
+    Base64DecodeError(#[from] DecodeError),
+    /// Error while decoding a base64 value from a slice.
+    #[error("There was an error while decoding a base64 value: `{0}`")]
```

Comment:
```suggestion
    #[error("There was an error while decoding a base64 value: {0}")]
    Base64DecodeError(#[from] DecodeError),
    /// Error while decoding a base64 value from a slice.
    #[error("There was an error while decoding a base64 value: {0}")]
```
Using backticks looks sorta unnatural when formatting the underlying error with `Display`